lib/httplib/httplib.go

```diff
@@ -150,7 +151,16 @@ func ReadJSON(r *http.Request, val interface{}) error {
 func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, error) {
 	if err != nil {
 		if uerr, ok := err.(*url.Error); ok && uerr != nil && uerr.Err != nil {
-			return nil, trace.ConnectionProblem(uerr.Err, uerr.Error())
+			// unescape any special characters from the URL (such as 'path%20foobar') to
+			// prevent MISSING from showing in connection problem logs
+			unescapedURL, err := url.QueryUnescape(uerr.URL)
+			if err != nil {
+				return nil, trace.ConvertSystemError(err)
+			}
+			// Replicates what uerr.Error() does but replaces the second argument of URL
+			// with our unescaped string instead
+			errorMessage := fmt.Sprintf("%s %q: %s", uerr.Op, unescapedURL, uerr.Err)
+			return nil, trace.ConnectionProblem(uerr.Err, errorMessage)
```

Comment:
I suggest avoiding placing the URL in the format string, or changing the URL to a different URL.

This would fix the bug and provide the behavior that (I think) was originally intended:
```go
return nil, trace.ConnectionProblem(uerr.Err, "%s %q: %s", uerr.Op, uerr.URL, uerr.Err)
```
That said, I'm not sure why it wasn't originally something like:
```go
return nil, trace.ConnectionProblem(err, "error parsing URL")
```

Reply:
I like your second option better. More concise and doesn't leave room for error. I wouldn't want to change the URL because then we'd be giving bad/wrong/slightly-off information which could lead to more confusion.